bin/ramalama

```diff
@@ -1,21 +1,47 @@
-#!/usr/bin/python3
+#!/usr/bin/env python3
 
-import argcomplete
+import glob
 import os
 import errno
 import subprocess
 import sys
 
+def get_pipx_dir():
+    python_version = f'{sys.version_info.major}.{sys.version_info.minor}'
+    argcompletion_path = os.path.expanduser(f'~/.local/pipx/venvs/argcomplete/lib/python{python_version}/site-packages')
+    if os.path.exists(argcompletion_path):
+        return argcompletion_path
+
+    # Use glob to find the matching directory for any Python version
+    path_pattern = '~/.local/pipx/venvs/argcomplete/lib/python*/site-packages'
+    matched_paths = glob.glob(path_pattern)
+
+    # Select the first matching path (if multiple matches are found, you can refine this)
+    if matched_paths and os.path.exists(matched_paths[0]):
+        return matched_paths[0]
 
 def main(args):
-    sharedirs = ["./", "/opt/homebrew/share/ramalama", "/usr/local/share/ramalama", "/usr/share/ramalama"]
+    sharedirs = ["./", "/opt/homebrew/share", "/usr/local/share", "/usr/share"]
     syspath = next((d for d in sharedirs if os.path.exists(d+"/ramalama/cli.py")), None)
     if syspath:
         sys.path.insert(0, syspath)
-    import ramalama
+
+    sys.path.insert(0, get_pipx_dir())
+
+    try:
+        import ramalama
+    except:
+        print(f"ramalama module not found in sys.path: {sys.path}", file=sys.stderr)
+        raise
 
     parser, args = ramalama.init_cli()
-    argcomplete.autocomplete(parser)
+
+    # if autocomplete doesn't exist, just do nothing, don't break
+    try:
+        import argcomplete
+        argcomplete.autocomplete(parser)
+    except Exception:
+        None
```

Comment:
We need a way to debug this, perhaps add a debugging level to RamaLama.

Reply:
SGTM, I'd like to catch this in a new PR though, really want to mimise the time macOS installs are broken.